tests/integration/test_registration_interface.py

```diff
@@ -42,3 +43,87 @@ def test_registration_interface(lwm2mserver, lwm2mclient):
+    text = lwm2mserver.waitfortime(20)
+    # Pass-Criteria B, D (C not observable)
+    assert text.find("lifetime: 300 sec") > 0 # NOTE: this should be 20 sec?
+    # expect update response immediately and again after every half "lifetime" (10s)
+    assert 2 <= text.count("Client #0 updated.") <= 3
+
+
+@pytest.mark.slow
+@pytest.mark.client_args("-t 60")
+def test_discarded_register_update(lwm2mserver, lwm2mclient):
+    """LightweightM2M-1.1-int-105
+    Test if Server replies correctly (with error message) upon receiving an
+    unexpected Register Update (e.g. after Lifetime expires). Test if Client
+    respects the error message and fallbacks full registration (re-register).
+
+    Test only partially implemented due to limitations (see comments)"""
+
+    lwm2mclient.waitfortext("STATE_READY")
+    # Test Procedure 1
+    # server learns client lifetime when client registers
+    # a server write to /1/x/1 does not affect the servers knowledge about
+    # client lifetime therefore a higher value than the inital registration lifetime
+    # is written to this resource to provoke the server to drop the client
+    text = lwm2mserver.waitfortime(60)
+    # Pass-Criteria A
+    assert text.find("Client #0 updated.")
+    # Test Procedure 2
+    # Client reports 60s lifetime to server
+    # Server changes client lifetime to 180
+    #   => server will drop client before an update is received
+    assert lwm2mserver.commandresponse("write 0 /1/0/1 180", "OK")
+    # ensure client is dropped >60s (in practice >30s)
+    text = lwm2mserver.waitfortime(70)
+    # Pass-Criteria B
+    assert text.find("Client #0 unregistered.") > 0
+    # Test Procedure 3
+    # wait long enough to client to send register update (180s total)
+    # Pass-Criteria C
+    # only in debug build it is possible to detect "4.04" from printouts
+    # Pass-Criteria D
+    # Client should perform full registration according as in int-101
+    # this is not implemented, client exits with "STATE_BOOTSTRAP_REQUIRED"
+
+
+@pytest.mark.slow
+#@pytest.mark.client_args("-t 60")
+def test_extending_lifetime_registration(lwm2mserver, lwm2mclient):
+    """LightweightM2M-1.1-int-107
+    Test that Server can extend the LwM2M lifetime of Client
+
+    Test only partially implemented due to limitations (see comments)"""
+
+    # Test Procedure 1
+    # Pass-Criteria A
+    lwm2mclient.waitfortext("STATE_READY")
+    # Test Procedure 2
+    assert lwm2mserver.commandresponse("write 0 /1/0/1 120", "OK")
+    text = lwm2mserver.waitforpacket()
+    # Pass-Criteria B
+    # a)
+    assert text.find("COAP_204_CHANGED") > 0
+    # b) NOT IMPLEMENTED
+    # c) not observable
+    # Test Procedure 3
+    text = lwm2mserver.waitfortime(120)
+    # Pass-Criteria C
+    # one update is sent immediately, expect implementation
+    # to send update at lifetime/2 intervals
+    # this does not seem to be the case for 120s (more frequent)
+    assert text.count("Client #0 updated.") >= 2
```

Comment:
Why not test for exactly two updates during the lifetime? This however interferes with #613 when testing against 120s.

Reply:
TC removed. (The exact number of updates is implementation dependent)